clientd3d/object.c

```diff
@@ -50,6 +50,30 @@ int CompareRoomObjectDistance(void *r1, void *r2)
 {
    return ((room_contents_node *) r1)->distance - ((room_contents_node *) r2)->distance;
 }
+/*************************************************************************/
+// Hierarchical compare objects - first on IsNumberObj then alphabetical
+//
+int CompareObjectNameAndNumber(void *obj1, void *obj2)
+{
+    // Cast obj1 and obj2 to the correct type
+   object_node *node1 = (object_node *)obj1;
+   object_node *node2 = (object_node *)obj2;
+
+   if (IsNumberObj(node1->id) != IsNumberObj(node2->id))
+   {
+      //Node2 before Node1 so that number object are on top
+      return (IsNumberObj(node2->id) - IsNumberObj(node1->id));
+   }
+   else
+   {
+      // Extract strings to compare
+      const char *string1 = LookupNameRsc(node1->name_res);
+      const char *string2 = LookupNameRsc(node2->name_res);
+
+      // Use stricmp to compare the name_res strings
```

Comment:
This comment simply mimics the code; remove it.

Reply:
Done.